cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1582,11 +1590,17 @@ void reader::impl::allocate_columns(read_mode mode, size_t skip_rows, size_t num
 
           // allocate
           // we're going to start null mask as all valid and then turn bits off if necessary
-          out_buf.create_with_mask(size, cudf::mask_state::ALL_VALID, _stream, _mr);
+          out_buf.create_with_mask(size, cudf::mask_state::UNINITIALIZED, _stream, _mr);
+          memset_bufs.push_back(cudf::device_span<uint8_t>((uint8_t *)(out_buf.data()), out_buf.data_size()));
+          nullmask_bufs.push_back(cudf::device_span<uint8_t>((uint8_t *)(out_buf.null_mask()), out_buf.null_mask_size()));
+
         }
       }
     }
   }
+  multibuffer_memset(memset_bufs, 0, _stream, _mr);
+  multibuffer_memset_validity(nullmask_bufs, 0xFF, _stream, _mr);
+  _stream.synchronize();
```

Comment:
```suggestion
```
This was just here for benchmarking purposes, right?  I don't think we want it here in general.